internal/translator/translate.go

```diff
@@ -157,10 +184,80 @@ func translateRoutes(irRoutes []irv1beta1.Route) ([]hpv1.Route, []hpv1.Include,
 			includes = append(includes, *hpInclude)
 			continue
 		}
-		route, translationWarnings := translateRoute(irRoute)
+		route, translationWarnings := translateRoute(irRoute, routeLCP)
 		routes = append(routes, route)
 		warnings = append(warnings, translationWarnings...)
 	}
 
 	return routes, includes, warnings
 }
+
+func extractPrefixes(routes []irv1beta1.Route) []string {
+
+	var prefixes []string
+	for _, route := range routes {
+		prefixes = append(prefixes, route.Match)
+	}
+
+	return prefixes
+}
+
+// longestCommonPathPrefix finds the longest common path prefix by
+// splitting a set of strings that give path prefixes on `/` characters,
+// then checking which match.
+func longestCommonPathPrefix(paths []string) string {
+
+	if len(paths) == 0 {
+		return ""
+	}
+
+	if len(paths) == 1 {
+		if paths[0] == "" || paths[0] == "/" {
+			return ""
+		}
+		return paths[0]
+	}
+
+	if !sort.StringsAreSorted(paths) {
+		sort.Strings(paths)
+	}
+
+	// Build a two-dimensional array of paths split by "/"
+	// the first element of pathElements will be the shortest path
+	pathElements := make([][]string, len(paths))
+	for index, path := range paths {
+		// Split the first '/' off, to remove the zero-length
+		// string that would otherwise be the first element.
+		if path[0] == '/' {
+			path = path[1:]
+		}
+		pathElements[index] = strings.Split(path, "/")
+	}
+
+	// Next, for each element in the shortest path,
+	// check if all the elements in that position in the other
+	// paths match. If so, it's common, add it.
+	// If not, that's it, break.
+	var longestPrefix []string
+	for index, pathElement := range pathElements[0] {
+		match := true
+		for _, pathSlice := range pathElements[1:] {
+			if pathSlice[index] != pathElement {
```

Comment:
The block at line 221 should ensure that paths is sorted, and so that pathElements is sorted. I think you referenced adding a test case to cover this, I'll do that.

Reply:
Yup, that makes sense.